python/ray/tune/execution/checkpoint_manager.py

```diff
@@ -100,10 +102,12 @@ def newest_persistent_checkpoint(self):
     @property
     def newest_checkpoint(self):
         """Returns the newest checkpoint (based on training iteration)."""
-        newest_checkpoint = max(
-            [self.newest_persistent_checkpoint, self.newest_memory_checkpoint],
-            key=lambda c: c.id,
-        )
+        checkpoints = [self.newest_memory_checkpoint, self.newest_persistent_checkpoint]
+        # Always prefer forced checkpoints
+        # If multiple are forced, take the one with the highest checkpoint id
+        checkpoints.sort(key=lambda c: c.id)
```

Comment:
I think we can use a single sort call here by using a tuple (`(c.id, c.forced)`), or even just keep `max` and use a tuple as key there